src/main/java/com/minecolonies/permissions/ColonyPermissionEventHandler.java

```diff
@@ -0,0 +1,232 @@
+     */
+    @SubscribeEvent
+    public void on(final PlayerInteractEvent event)
+    {
+        if (colony.isCoordInColony(event.getWorld(), event.getPos()))
+        {
+            final Block block = event.getWorld().getBlockState(event.getPos()).getBlock();
+            // Huts
+            if (block instanceof AbstractBlockHut &&
+                  !colony.getPermissions().hasPermission(event.getEntityPlayer(), Permissions.Action.ACCESS_HUTS))
+            {
+                cancelEvent(event);
+            }
+        }
+    }
+
+    /**
+     * Check Permissions for Container Open Events.
+     *
+     * @param event the event to check on.
+     */
+    @SubscribeEvent
+    public void on(final PlayerContainerEvent.Open event)
+    {
+        if (this.colony.isCoordInColony(event.getEntity().getEntityWorld(), event.getEntity().getPosition())
+              && !this.colony.getPermissions().isColonyMember(event.getEntityPlayer()))
+        {
+            cancelEvent(event);
+        }
+    }
+
+    /**
+     * ItemTossEvent handler.
+     *
+     * @param event ItemTossEvent
+     */
+    @SubscribeEvent
+    public void on(ItemTossEvent event)
+    {
+        final EntityPlayer playerIn = event.getPlayer();
+        if (colony.isCoordInColony(playerIn.getEntityWorld(), playerIn.getPosition()))
+        {
+            Permissions.Rank rank = colony.getPermissions().getRank(playerIn);
+
+            if (rank.ordinal() < Permissions.Rank.NEUTRAL.ordinal())
+            {
+                /*
+                    this will delete the item entirely:
+                    Canceling the event will stop the items from entering the world,
+                    but will not prevent them being removed from the inventory
+                    - and thus removed from the system.
+                 */
+                cancelEvent(event);
+            }
+        }
+    }
+
+    /**
+     * Template
+     */
+    public void on()
+    {
+        //
+    }
+
+    /**
+     * This method returns TRUE if this event should be denied.
+     *
+     * @param worldIn    the world to check in
+     * @param posIn      the block to check
+     * @param playerIn   the player who tries
+     * @param blockState the state that block is in
+     * @return true if canceled
+     */
+    private boolean checkBlockEventDenied(final World worldIn, final BlockPos posIn, final EntityPlayer playerIn, final IBlockState blockState)
+    {
+        if (colony.isCoordInColony(worldIn, posIn))
+        {
+            if (colony.getPermissions().isColonyMember(playerIn))
+            {
+                final Permissions.Rank rank = colony.getPermissions().getRank(playerIn);
+                if (rank.ordinal() >= Permissions.Rank.NEUTRAL.ordinal())
+                {
+                    return true;
+                }
+                if (blockState.getBlock() instanceof AbstractBlockHut)
+                {
+                    if (rank.ordinal() >= Permissions.Rank.OFFICER.ordinal())
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1066)